ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1555,6 +1839,10 @@ FindInfo findDeletedEntriesSince(FindToken token, long maxTotalSizeOfEntries, lo
         }
       }
     }
+    // Filter out all the messages that are not "deleted", then update state for remaining deleted message.
+    filterDeleteEntries(messageEntries);
+    updateStateForMessages(messageEntries);
+
     filterDeleteEntries(messageEntries);
```

Comment:
why here we filter delete entries again? Is this because messageInfo is updated in `updateStateForMessages`?

Reply:
talked offline